content/blog/2022-06-14-DVC-VS-Code-extension.md

```diff
@@ -0,0 +1,135 @@
+date: 2022-06-14
+description: >
+  Streamline your data science workflow with the DVC extension for VS Code. Data
+  management and experiments all from the comfort of your IDE!
+descriptionLong: >
+  VS Code has become the IDE of choice for many data scientists. Today we are
+  releasing the DVC extension, which will streamline your data science workflow
+  even further. Manage your data, run experiments, compare metrics, and
+  visualize plots, all from the comfort of your IDE.
+# picture: 2022-06-14/header.png
+author: rob_dewit
+# commentsUrl:
+tags:
+  - VS Code
+  - DVC
+  - MLOps
+  - Release
+---
+
+Since its beta release in 2017 DVC has grown to be an essential tool for many
+data science teams. Its data versioning capabilities, reproducible pipelines,
+and experiment tracking features are at the core of our ecosystem of open MLOps
+tools.
+
+Today we are proud to launch a new product that extends the ways in which ML
+teams can use DVC: our extension for Visual Studio Code.
+
+With the VS Code extension, you can control your datasets, run experiments, view
+metrics and plots, and much more. All from your IDE.
+
+[Install the DVC extension from the marketplace to get started.](https://marketplace.visualstudio.com/items?itemName=Iterative.dvc)
+
+# Why an extension?
+
+DVC was built to expand upon the Git workflow. This approach has made it a
+powerful tool that can easily be picked up by users familiar with Git. However,
+for those less familiar with CLI tools, it can be more difficult to get
+accustomed to using DVC. Moreover, while DVC provides many useful
+visualizations, a CLI tool can only take things so far.
+
+With this extension we want to achieve the following:
+
+- Make DVC more accessible by hiding the complexity of the CLI
+- Enhance existing visualizations and providing new ones
+- Move the entire data science workflow into your IDE
+
+# Why Visual Studio Code
+
+One of [our core beliefs at Iterative](https://iterative.ai/why-iterative/) is
+that our products should couply tightly with existing tools and processes for
+data science projects. VS Code is the IDE of choice for many of us at Iterative,
+and we really like it. Here's a few reasons why:
+
+VS Code...
+
+- is open source and lightweight
+- is becoming an industry standard for Python and data science
+- has great support for Git workflows
+- supports Jupyter Notebooks
+- comes with remote development and execution built in
+
+# Features
+
+Our first course of action was to include all DVC features related to data
+management and experiments in the extension. Here are the main features that are
+available when you install the extension right now:
+
+## Command palette
+
+The DVC extension makes extensive use of the Command Palette, reducing the need
+to learn commands by heart:
+
+![Use the command palette to quickly find DVC
+features](/uploads/images/2022-06-14/command-palette.gif)
+
+## Source control management
```

Comment:
Let's start with the experiments table, how can you run and bookkeep experiments - that's the main value prop, SCM / data files - I would may be mention that it's also included but do the full description at the end. It's more like a feature needed to run the experiments.

Plots should also come before data - ability to visualize, to run and follow the metrics, etc

Reply:
Can we also keep it close to "DVC tracked explorer" or combine them? It's unclear the different value of each. We don't need to address every pane/view for the blog post.

Reply:
Moved it around; current order:

- DVC View: experiments, parameters, metrics, and plots
- Source control management
- DVC tracked explorer
- Command palette

Reply:
- SCM / DVC tracked should come together - it's about managing datasets, artifacts, giving a simpler interface to deal with data
- Command palette - no need to do a separate section. You could mention about it in the intro to one of the sessions ... it's not an independent large feature at all